src/com/boxboat/jenkins/library/deploy/DeployTarget.groovy

```diff
@@ -0,0 +1,11 @@
+package com.boxboat.jenkins.library.deploy
+
+import com.boxboat.jenkins.library.config.BaseConfig
+
+class DeployTarget extends BaseConfig<Deployment> implements Serializable  {
```

Comment:
This class name is not good, there is already `IDeployTarget` and this `DeployTarget` does not implement `IDeployTarget`.  I would call this `DeployTargetRef`